extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -547,6 +548,10 @@ class QueueProcessor extends EventEmitter {
                               { error: sourceEntry.error });
             return process.nextTick(() => done(errors.InternalError));
         }
+        if (sourceEntry.skip) {
+            // skip message, noop
```

Comment:
I would add a debug log here as it can be useful if there is any concern with canary processing.

Reply:
there will be a lot of canary messages and I wanted to keep them silent because all consumer groups for replication topic will receive these messages. When a new storage location is added, it will send off more canaries, and all these consumers will receive again. Theres definitely room for improvement

Reply:
As long as messages do not go at a crazy rate like 10+ messages per second, it's good to be able to track what is going on, that is what debug logs are for. Imagine a race condition prevents the message from being received on time, or some issue with timing, it will be hard to diagnose without logs showing exactly when messages have been consumed. On kafka we can fetch the message timestamps to know when messages have been published.

I think there is value in having a debug log here, but if you really think this will be too much logging, you can leave as is.

Reply:
I have an issue here where we will have multiple pods, so each new queueprocessor on each pod will send the canary, so there will be duplicate entries sent. Since I get the kafka topic md and send for all existing partitions, each consumer will send a msg to each partition. I haven't been able to figure out a way to do this per consumer on a partition because a consumer may subscribe to more than one partition if one or more consumers never connect, or if we have more partitions than consumers. So again, room for improvements here